Crash recovery should deal better with corrupt data for individual series in checkpoint file

**What did you do?**

upgrade to newer version or prometheus

**What did you expect to see?**

clean start

**What did you see instead? Under which circumstances?**

crash

**Environment**

```
WARN[0007] Lost at least 9 chunks for fingerprint de012972ecd6f680, metric tunein_taps{country="DE", instance="localhost:9301", item="s236774", job="prometheus", source="browse"}.  source=crashrecovery.go:116
panic: runtime error: makeslice: cap out of range

goroutine 1 [running]:
panic(0x18b2120, 0xc446690260)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/prometheus/prometheus/storage/local.(*persistence).recoverFromCrash(0xc420362c60, 0xc4200a90b0, 0x1, 0x17bdd20)
        /go/src/github.com/prometheus/prometheus/storage/local/crashrecovery.go:119 +0xb82
github.com/prometheus/prometheus/storage/local.(*persistence).loadSeriesMapAndHeads.func1(0xc420362c60, 0xc445e859e8, 0xc4200a90b0, 0xc445e859d8)
        /go/src/github.com/prometheus/prometheus/storage/local/persistence.go:817 +0xf6
github.com/prometheus/prometheus/storage/local.(*persistence).loadSeriesMapAndHeads(0xc420362c60, 0xc42000e500, 0x0, 0x267d540, 0xc420010190)
        /go/src/github.com/prometheus/prometheus/storage/local/persistence.go:840 +0x3f5
github.com/prometheus/prometheus/storage/local.(*MemorySeriesStorage).Start(0xc420238b00, 0x0, 0x0)
        /go/src/github.com/prometheus/prometheus/storage/local/storage.go:374 +0x1f5
main.Main(0x0)
        /go/src/github.com/prometheus/prometheus/cmd/prometheus/main.go:181 +0x114f
main.main()
        /go/src/github.com/prometheus/prometheus/cmd/prometheus/main.go:43 +0x22
```

* System information:

uname -srm
Linux 4.4.0-62-generic x86_64

* Prometheus version: 

# ./prometheus --version
prometheus, version 1.5.2 (branch: master, revision: bd1182d29f462c39544f94cc822830e1c64cf55b)
  build user:       root@a8af9200f95d
  build date:       20170210-14:41:22
  go version:       go1.7.5

* Alertmanager version:

	insert output of `alertmanager -version` here (if relevant to the issue)

* Prometheus configuration file:
```
insert configuration here
```

--- a/storage/local/crashrecovery.go
+++ b/storage/local/crashrecovery.go
@@ -86,11 +86,12 @@ func (p *persistence) recoverFromCrash(fingerprintToSeries map[model.Fingerprint
 	for fp, s := range fingerprintToSeries {
 		if _, seen := fpsSeen[fp]; !seen {
 			// fp exists in fingerprintToSeries, but has no representation on disk.
-			if s.persistWatermark == len(s.chunkDescs) {
+			if s.persistWatermark >= len(s.chunkDescs) {
 				// Oops, everything including the head chunk was
-				// already persisted, but nothing on disk.
-				// Thus, we lost that series completely. Clean
-				// up the remnants.
+				// already persisted, but nothing on disk. Or
+				// the persistWatermark is plainly wrong. Thus,
+				// we lost that series completely. Clean up the
+				// remnants.
 				delete(fingerprintToSeries, fp)
 				if err := p.purgeArchivedMetric(fp); err != nil {
 					// Purging the archived metric didn't work, so try

